hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/tools/DFSAdmin.java

```diff
@@ -201,8 +201,19 @@ public void run(Path path) throws IOException {
       super(conf);
       CommandFormat c = new CommandFormat(2, Integer.MAX_VALUE);
       List<String> parameters = c.parse(args, pos);
-      this.quota =
-          StringUtils.TraditionalBinaryPrefix.string2long(parameters.remove(0));
+      String str = parameters.get(0).trim();
+      try {
+        this.quota = StringUtils.TraditionalBinaryPrefix
+                .string2long(parameters.remove(0));
+      } catch(NumberFormatException e){
+        throw new IllegalArgumentException("\"" + str +
+                "\" is not a valid value for a quota.");
+      }
+      if (HdfsConstants.QUOTA_DONT_SET == this.quota) {
+        System.out.print("WARN: \"" + this.quota +
```

Comment:
Changed to ```System.err.print```.  And I agree with you not need to execute the backward code, but here I confused is this message just a warning message, it is suitable to throw an Exception ? Any ideas? Thanks!

Reply:
or how about return directly? I think it is not necessary to send RPC request to NameNode if has checked the parameter is invalid.

Reply:
I tried return directly before, but there is an extra invalid message throw ```setQuota: null```.  checked the code it is not easy to change, because the warn message is in the ```SetSpaceQuotaCommand``` constructor. Maybe throw an exception or current solution are both OK. any ideas? Thanks!

Reply:
If that I think it is OK to throw exception that avoid send RPC request to NameNode if parameter is invalid which client has checked.

Reply:
Agree with U, change to throw exception. Thanks for comfirm!